chopper.py

```diff
@@ -229,189 +268,193 @@ def split_by_columns(filepath: Path, col_list: list[str], config: Config) -> lis
-        for i, row in enumerate(f):
-            if i == 0:
-                headers = row
-                continue
-
-            # Initialize new file.
-            if (i - 1) % rows == 0:
-                filenum += 1
-                tmp_file = config.output_dir / f"{filepath.stem}_{filenum}"
-                files.append(tmp_file)
-                fout = open(tmp_file, "w", encoding=config.encoding)
-                fouts.append(fout)
-                fout.write(headers)
+    with filepath.open("rb", encoding=config.encoding) as fin_obj:
+        with mmap.mmap(fin_obj.fileno(), length=0, **config.mmap_kwargs) as fin:
+            if os.name != "nt":
+                fin.madvise(mmap.MADV_SEQUENTIAL)
 
-            fout.write(row)
+            headers = fin.readline()
+            filenum = 0
+            paths = []
+            fout = None
+            i = 0
 
-    for fout in fouts:
-        fout.close()
+            while row := fin.readline():
+                # Initialize new file.
+                if i % rows == 0:
+                    if fout:
+                        fout.close()
+                    filenum += 1
+                    tmp_path = config.output_directory / f"{filepath.stem}_{filenum}"
+                    paths.append(tmp_path)
+                    fout = tmp_path.open("wb", encoding=config.encoding)
+                    fout.write(headers)
 
-    if not config.is_original:
-        os.remove(filepath)
+                fout.write(row)
+                i += 1
 
-    return files
+    if filepath.absolute().parents[0] == config.output_directory.absolute():
+        filepath.unlink()
 
+    return paths
 
-def combine_files(in_files: list[Path], config: Config) -> Path:
+
+def combine_files(in_paths: list[Path], config: ChopperNamespace) -> Path:
     """Combines multiple files into one.
 
     Args:
-        in_files (list[Path]): List of files to combine.
-        config (Config): Config object created from CLI arguments.
+        in_paths (list[Path]): List of files to combine.
+        config (ChopperNamespace): Namespace object created from CLI arguments.
 
     Returns:
         Path: Path of the combined file.
     """
 
-    if len(in_files) == 1:
-        return in_files[0]
+    if len(in_paths) == 1:
+        return in_paths[0]
 
-    combined_fp = config.output_dir / "combined"
-    with open(combined_fp, "w", encoding=config.encoding) as fout:
-        for i, f in enumerate(in_files):
-            with open(f, "r") as fin:
+    combined_fp = config.output_directory / "combined"
+    with combined_fp.open("wb", encoding=config.encoding) as fout:
+        for i, f in enumerate(in_paths):
+            with f.open("rb", encoding=config.encoding) as fin:
                 if i > 0:  # Skip the header row, except for the first file.
-                    next(fin)
-                for line in fin:
-                    fout.write(line)
+                    fin.readline()
+                shutil.copyfileobj(fin, fout)
```

Comment:
shutil's copy operations are highly optimized and memory-friendly already. Big gains here in file combination time.